### PR TITLE
Feat/education experience update

### PR DESCRIPTION
**Pull request by nehamamandelbaum**
This PR completes issue #563, which was to implement the update method for education experiences. 

How the UI looks: 

Captura de Tela 2023-10-31 às 15 06 11
Captura de Tela 2023-10-31 às 15 06 25
Captura de Tela 2023-10-31 às 15 06 40

### CODE REVIEW

File: app/controllers/education_experiences_controller.rb [hunk 1]
```diff
@@ -1,15 +1,14 @@
 # frozen_string_literal: true
 
 class EducationExperiencesController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound do
+    redirect_to '/404'
+  end
```
Comment (adbatista) [inline app/controllers/education_experiences_controller.rb:6]: this is repeated in various places, could be moved to application controller

File: app/decorators/education_experience_pagination_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(params, object, options = {})
+    @params = params
+    @unpaginated_object = object
+    @paginated_object = @unpaginated_object
+                        .page(@params[:page])
+                        .per(@params[:per])
+
+    super(@paginated_object, options)
+  end
+end
```
Comment (adbatista) [inline app/decorators/education_experience_pagination_decorator.rb:15]: is this used

File: app/models/education_experience.rb [hunk 3]
```diff
@@ -14,4 +14,6 @@ class EducationExperience < ApplicationRecord
                           message: :less_than_start_date,
                           greater_than: ->(education_experience) { education_experience.start_date },
                           allow_nil: true
+
+  scope :for_user, ->(user_id) { where(user_id:) }
```
Comment (adbatista) [inline app/models/education_experience.rb:18]: this is not being used

File: app/views/education_experiences/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    <div class="col-lg-8">
+      <%= f.input :institution, as: :string, placeholder: 'Universidade de São Paulo' %>
+      <%= f.input :course, as: :string, placeholder: 'Computer Science' %>
+      <%= f.hidden_field :user_id, value: current_user.id %>
+      <%= f.input :start_date, as: :string, input_html: { class: 'datepicker', data: { allow_overtime: 'true' } } %>
+      <%= f.input :end_date, as: :string, input_html: { class: 'datepicker', data: { allow_overtime: 'true' }, required: false  } %>
+    </div>
+  </div>
+
+<%= f.submit t('.save') %>
```
Comment (adbatista) [inline app/views/education_experiences/_form.html.erb:12]: ident missing

File: app/views/refills/_navigation.html.erb [hunk 5]
```diff
@@ -52,6 +52,21 @@
...
 
+        <div class="links">
+          <div class="dropdown">
+            <%= link_to "#", data: {toggle: "dropdown"}, class: "dropdown-toggle resume" do %>
+              <%= t('.my_resume') %><span class="caret"></span>
+            <% end %>
+            <ul class="dropdown-menu bg-danger" style="min-width: 14rem" role="menu" aria-labelledby="dLabel" style="color:#c61515">
+              <li>
+                <%= link_to education_experiences_path, class:"justify-content-start" do %>
+                  <%= t('activerecord.models.education_experience.one') %>
```
Comment (adbatista) [inline app/views/refills/_navigation.html.erb:63]: you can use Model.model_name... here

File: app/views/refills/_navigation.html.erb [hunk 6]
```diff
@@ -52,6 +52,21 @@
           <% end %>
         </div>
 
+        <div class="links">
+          <div class="dropdown">
+            <%= link_to "#", data: {toggle: "dropdown"}, class: "dropdown-toggle resume" do %>
+              <%= t('.my_resume') %><span class="caret"></span>
+            <% end %>
+            <ul class="dropdown-menu bg-danger" style="min-width: 14rem" role="menu" aria-labelledby="dLabel" style="color:#c61515">
```
Comment (adbatista) [inline app/views/refills/_navigation.html.erb:60]: avoid style attributes, use css here

File: spec/controllers/education_experiences_controller.rb [hunk 7]
```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe EducationExperiencesController, type: :controller do
+  render_views
+
+
```
Comment (adbatista) [inline spec/controllers/education_experiences_controller.rb:8]: remove extra spaces

File: spec/controllers/education_experiences_controller.rb [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        get :index
+
+        expect(response).to have_http_status(:ok)
+      end
+
+      context 'when the user does not contain education experiences' do
+        it 'renders the table with not found message' do
+          get :index
+
+          expect(page.find('table')).to have_content(I18n.t('resumes.education_experience.not_found'))
```
Comment (adbatista) [inline spec/controllers/education_experiences_controller.rb:36]: use the expected text here instead the translation helper

File: spec/features/dashboard_spec.rb [hunk 9]
```diff
@@ -3,6 +3,23 @@
...
+
+    context 'when the user is logged in' do
+      before do
+        sign_in user
+      end
+
+      it 'has the dropdown option on the navigation bar' do
+        visit root_path
+
+        # find('.resume').click
```
Comment (adbatista) [inline spec/features/dashboard_spec.rb:17]: remove dead code

**Review by adbatista — COMMENTED** [review]
resolve conflicts please

File: app/views/education_experiences/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<ul class="breadcrumb">
+  <li li class="breadcrumb-item"><%= link_to t(:back, scope: :helpers), education_experiences_path %></li>
```
Comment (EduardoSCosta) [inline app/views/education_experiences/new.html.erb]: The same thing as on the edit page, and remove this "li" attribute.

File: app/views/education_experiences/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+</ul>
+
+<h1 class="education-experience-form-title">
```
Comment (EduardoSCosta) [inline app/views/education_experiences/new.html.erb]: Same thing for this one.

File: app/views/education_experiences/edit.html.erb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<ul class="breadcrumb">
+  <li class="breadcrumb-item"><%= link_to t(:back, scope: :helpers), education_experiences_path %></li>
```
Comment (EduardoSCosta) [inline app/views/education_experiences/edit.html.erb:2]: It looks like this class isn't applying any styles:
image

If it's not being used for anything else, you can simply remove it:
```suggestion
  <li><%= link_to t(:back, scope: :helpers), education_experiences_path %></li>
```

File: app/views/education_experiences/edit.html.erb [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+</ul>
+
+<h1 class="education-experience-form-title">
```
Comment (EduardoSCosta) [inline app/views/education_experiences/edit.html.erb:5]: Same thing for this one.

File: spec/controllers/education_experiences_controller.rb [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+        end
+      end
+
+      context 'when the user has educatione experiences' do
```
Comment (EduardoSCosta) [inline spec/controllers/education_experiences_controller.rb]:
```suggestion
      context 'when the user has educational experiences' do
```

**Review by adbatista — APPROVED** [review]

**Pull request merged**